filament/backend/src/webgpu/WebGPUDriver.h

```diff
@@ -107,6 +105,11 @@ class WebGPUDriver final : public DriverBase {
         return mHandleAllocator.handle_cast<D*>(handle);
     }
 
+    template<typename B, typename D,
+            typename = std::enable_if_t<std::is_base_of_v<B, D>, D>>
+    void destruct(Handle<B>& handle, D const* p) noexcept {
+        return mHandleAllocator.deallocate(handle, p);
+    }
```

Comment:
ii was thinking that this could be something like 
```suggestion
       template<typename D, typename B>
    void destructHandle(Handle<B>& handle) noexcept {
        auto* p = mHandleAllocator.handle_cast<D*>(handle);
        mHandleAllocator.deallocate(handle, p);
    }
```
which is from the metal driver